### PR TITLE
Improve AWS deployments (and a couple other things)

### PR DESCRIPTION
**Pull request by smondet**
Not yet 100% ready but it's better to merge progressively into master.

### CODE REVIEW

File: src/app/make_dockerfiles.ml [hunk 1]
```diff
@@ -493,8 +493,8 @@ module Image = struct
           Test.test_dashdashversion "ketrew" "3.1.0+dev";
           ketrew_with_pg;
           (* coclobas --version gives: "0.0.2-dev+<commit-hash-prefix>": *)
-          Test.succeeds "coclobas --version | grep 0.0.2-dev";
-          Test.succeeds "ocamlfind list | grep coclobas | grep 0.0.2-dev";
+          Test.succeeds "coclobas --version | grep 0.0.3-dev";
```
Comment (armish) [inline src/app/make_dockerfiles.ml:496]: maybe time for this to be extracted as a variable ;)

**Review by armish — APPROVED** [review]
LGTM as far as I understand:)

Did two passes but nothing apparently wrong with it. I am not that familiar with the Amazon's infrastructure, so let me know when you need somebody to test it out and that way I can have a better sense of all the components, too

File: src/app/make_dockerfiles.ml (hunk 1 above)
Comment (smondet) [reply src/app/make_dockerfiles.ml:496]: I actually like having those hard coded and going through the notice-failure/fix/rebuild process :)
that way the code says exactly which tests had to pass before pushing to `hub.docker.com`

File: src/app/example_aws.ml [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+          "Optional users and passwords, if you want the authentication proxy.\n\
+           Generate this with the `htpasswd` utiliy or with \n\
+           <http://aspirine.org/htpasswd_en.html>\n\
+           Please use the `bcrypt` hash (others may not work with nginx by default)";
+    ];
+    section "DNS / TLS" [
+      env "certificate_email" ~required:true ~example:"sherlock@example.com"
+        ~help:"Let's Encrypt wants an email address to associate with\
+               the TLS certificate";
+    ];
```
Comment (smondet) [inline src/app/example_aws.ml]: Forgot to remove this, let's encrypt does not work on `*.amazonaws.com`

**Pull request merged**